tests/e2e/vmlistener_test.go

```diff
@@ -80,41 +74,77 @@ func (s *VMListenerTestParams) TearDownTest(c *C) {
 		out, err := dockercli.RemoveContainer(s.vm1, s.containerName)
 		c.Assert(err, IsNil, Commentf(out))
 	}
-	out, err := dockercli.DeleteVolume(s.vm1, s.volumeName)
-	c.Assert(err, IsNil, Commentf(out))
 }
 
 var _ = Suite(&VMListenerTestParams{})
 
 // Test vmdkops service restart
-// 1. Setup: Create a volume
-// 2. Setup: Attach it to a container
-// 3. Restart the vmdkops service
-// 4. Verification: volume stays as attached
+// 1. Create a volume
+// 2. Attach the volume to a container
+// 3. Verification: volume is in attached status
+// 4. Restart the vmdkops service
+// 5. Verification: volume stays as attached
+// 6. Remove the container
+// 7. Remove the volume
 func (s *VMListenerTestParams) TestServiceRestart(c *C) {
 	misc.LogTestStart(c.TestName())
 
+	// Create the volume
+	out, err := dockercli.CreateVolume(s.vm1, s.volumeName)
```

Comment:
I don't think so. Only truly common logic should be handled by SetUp/TearDown. For some test groups, such as vsan_test, basic_test, swarm_test, it's not the case. The reason I move the logic to test itself is because it's not common for all tests in this group.

I don't think it's a good idea to create separate test groups based on the common logic in SetUp/TearDown. Instead, test group should be created based on different test scenarios.

We can sync up offline if you have further comments.